src/daemon/mod.rs

```diff
@@ -171,11 +161,12 @@ pub(super) async fn start(
 
     let keystore = Arc::new(RwLock::new(keystore));
 
-    let chain_data_path = chain_path(&config);
-    let db = Arc::new(ManyCar::new(Arc::new(open_proxy_db(
-        db_root(&chain_data_path),
-        config.db_config().clone(),
-    )?)));
+    if opts.exit_after_init {
+        return Ok(());
+    }
```

Comment:
Why did you move this?

Reply:
All `daemon` integration tests rely on this flag to not fail because they don't provide a snapshot. I have to either move this early or do `process::exit(0)` when there's no permission to download a snapshot, see https://github.com/ChainSafe/forest/pull/3365#discussion_r1299220428

Reply:
Oh, I see. You had to move this check earlier because you moved the snapshot downloading into the DB opening code. Hmm. We can clean things up once we've refactored the DB opening code.

Reply:
Moved back